validator/validators/performance/performance_validator.go

```diff
@@ -45,19 +46,34 @@ func NewPerformanceValidator(vConfig models.ValidateConfig) models.ValidatorFact
 }
 
 func (s *PerformanceValidator) CheckData(startTime, endTime time.Time) error {
-	metrics, err := s.GetPerformanceMetrics(startTime, endTime)
-	if err != nil {
-		return err
-	}
+	if strings.Contains(s.vConfig.GetTestCase(), "windows") {
+		stat, err := s.GetWindowsPerformanceMetrics(startTime, endTime)
+		if err != nil {
+			return err
+		}
+		statInfo, err := s.CalculateWindowsMetricStatsAndPackMetrics(stat)
+		if err != nil {
+			return err
+		}
+		err = s.SendPacketToDatabase(statInfo)
+		if err != nil {
+			return err
+		}
+	} else {
+		metrics, err := s.GetPerformanceMetrics(startTime, endTime)
+		if err != nil {
+			return err
+		}
 
-	perfInfo, err := s.CalculateMetricStatsAndPackMetrics(metrics)
-	if err != nil {
-		return err
-	}
+		perfInfo, err := s.CalculateMetricStatsAndPackMetrics(metrics)
+		if err != nil {
+			return err
+		}
 
-	err = s.SendPacketToDatabase(perfInfo)
-	if err != nil {
-		return err
+		err = s.SendPacketToDatabase(perfInfo)
```

Comment:
If this is common for both, can sit outside the if/else logic.

Reply:
Implemented this in the new revision